src/app/core/services/download.service.ts

```diff
@@ -10,6 +12,18 @@ import { DownloadProgress } from '../../../../shared/models/download.model';
 import { FileUnzipEvent } from '../../../../shared/models/unzip.model';
 import { Mod } from '../models/mod';
 import { UserSettingModel } from '../../../../shared/models/user-setting.model';
+import { environment } from '../../../environments/environment';
+
+export interface ModData {
```

Comment:
But if we only need the name, version and link, you can use a interface, like models/mod.ts

Reply:
I trimmed the interface down to just name, version and link, because I failed to reuse the one in mod.ts, I wasn't able to map the json without renaming the properties of the Mod interface. 😢 

The download will always fallback to headless solution if the indexed mod list is empty or target mod is not found in it.
But let's hope they keep the JSON or maybe even extend it. Not sure why they wouldn't support it since it's so lightweight.